assets/reviews.js

```diff
@@ -125,6 +125,224 @@ const setupReviews = async () => {
+    radio.addEventListener('change', function() {
+      const starCountSpan = document.querySelector('.review-stars-count');
+      starCountSpan.textContent = `(${this.value} ${ratings})`;
+    });
+  });
+
   setupReviews();
-});
+}
+
+async function handleReviewFormSubmit(e) {
+  e.preventDefault();
+  clearFieldErrors();
+
+  const formData = new FormData(e.target);
+  Object.assign(reviewData, {
+    content: sanitizeInput(formData.get('content')),
+    email: formData.get('email'),
+    ratings: Number(formData.get('ratings')),
+    first_name: formData.get('first_name'),
+    last_name: formData.get('last_name')
+  });
+
+  try {
+      const response = await youcanjs.product.submitReview(reviewsProductId, reviewData);
+      if (response) {
+        notify(`${REVIEWS_TRANSLATED_TEXT.successMessage}`, 'success');
+        e.target.reset();
+        e.target.style.display = 'none';
+        document.querySelector('.thank-you-message').style.display = 'flex';
+        document.querySelector('.modal-title').style.display = 'none';
+      } else {
+        notify(`${REVIEWS_TRANSLATED_TEXT.errorMessage}`, 'error');
+      }
+  } catch (error) {
+      handleReviewError(error);
+  }
+}
+
+function handleReviewError(error) {
+  notify(error.message, 'error');
+  if (error) {
+      for (const field in error.meta.fields) {
+        const errorMsg = error.meta.fields[field][0];
+        displayFieldError(field, errorMsg);
+      }
+  } else {
+    notify(`${REVIEWS_TRANSLATED_TEXT.errorMessage}`, 'error');
+  }
+}
+
+function displayFieldError(fieldName, errorMsg) {
+  const inputElement = document.querySelector(`[name="${fieldName}"]`);
+  if (inputElement) {
+      const errorElement = document.createElement('span');
+      errorElement.className = 'field-error';
+      errorElement.textContent = errorMsg;
+      inputElement.parentElement.insertBefore(errorElement, inputElement.nextSibling);
+  }
+}
+
+function clearFieldErrors() {
+  const errorElements = document.querySelectorAll('.field-error');
+  errorElements.forEach(el => el.remove());
+}
+
+function showModal() {
+  const modal = document.getElementById("reviewModal");
+  modal.style.display = "flex";
+  document.body.style.overflow = 'hidden';
+}
+
+function hideModal() {
+  const modal = document.getElementById("reviewModal");
+  modal.style.display = "none";
+  document.body.style.overflow = '';
+}
+
+function sanitizeInput(input) {
+  const div = document.createElement('div');
+  div.innerHTML = input;
+  return div.textContent || div.innerText || "";
+}
+
+function uploadReviewImage(container, event) {
+  const targetElement = event.target;
+
+  const imgElement = container.querySelector('.uploaded-image');
+  if (imgElement && imgElement.style.display === 'block' && targetElement !== container.querySelector('.add-more')) {
+      return;
+  }
+
+  const uploadInput = createUploadInput();
+  uploadInput.addEventListener('change', handleFileChange);
```

Comment:
you need to set a param for handleFileChange function since we also have upload image input **Variant**